File: webclient/comp/display.js
function DisplayLMModule (root) {
		
	var padding = "00000000";
	var textWindow = root.find("#textWindow");
	
	function defaultLine(text) {
		displayLine({text: text, color: 0});
		scroll();
	};
	
	function displayLine(line) {
		var htmlline = $("<span>" + line.text + "</span>");
		var color = parseInt(line.color).toString(16).toUpperCase();
		color = padding.substring(0, 6-color.length) + color;
		htmlline.attr("style", "color: #" + color);
		textWindow.append(htmlline);
		textWindow.append($("<br />"));
	};

	function scroll () {
		textWindow.scrollTop(textWindow[0].scrollHeight);
	};

	function display(display) {
		lines = display.lines;
		for (var i = 0; i < lines.length; i++) {
			displayLine(lines[i]);
		}
		scroll();		
	};
		
	this.register("display", display);
	this.register("default_line", defaultLine);
	lmdp.dispatch("display_ready");
}

DisplayLMModule.prototype = LMWeb.BaseModule;

File: lampost/dto/display.py
from rootdto import RootDTO

class Display(RootDTO):
    def __init__(self, text=None, color=0x000000):
        self.display = RootDTO()
        self.display.lines = []
        if text is not None:
            self.append_line(DisplayLine(text, color))

    def merge(self, other):
        for line in other.display.lines:
            self.append_line(line)
        return self

    def append_line(self, line):
        self.display.lines.append(line)
        return self

    def append_text(self, text, color=0x000000):
        self.display.lines.append(DisplayLine(text, color))
        return self

    @property
    def feedback(self):
        return self

class DisplayLine(RootDTO):
    def __init__(self, text, color=0x000000):
        self.text = text
        self.color = color



